.drone.yml

```diff
@@ -0,0 +1,231 @@
+
+- name: audit
+  image: node:lts
+  commands:
+  - npm audit
+
+---
+kind: pipeline
+name: lint
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: build information
+  image: node:lts
+  commands:
+  - node --version
+  - npm --version
+
+- name: lint
+  image: node:lts
+  commands:
+  - "npx eslint 'src/**/*.js'"
+
+---
+kind: pipeline
+name: unit tests (node:8)
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: build information
+  image: node:8
+  commands:
+  - node --version
+  - npm --version
+
+- name: install
+  image: node:8
+  commands:
+  - npm install
+
+- name: unit tests
+  image: node:8
+  commands:
+  - npm run test-node
+  environment:
+    BITGOJS_TEST_PASSWORD:
+      from_secret: password
+
+- name: upload coverage
+  image: node:8
+  commands:
+  - "npx nyc report --reporter=text-lcov > coverage.lcov"
+  - "npx codecov -f coverage.lcov -B \"${DRONE_BUILD_NUMBER:-0}\" -t \"$CODECOV_TOKEN\""
+  environment:
+    CODECOV_TOKEN:
+      from_secret: codecov
+
+---
+kind: pipeline
+name: unit tests (node:9)
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: build information
+  image: node:9
+  commands:
+  - node --version
+  - npm --version
+
+- name: install
+  image: node:9
+  commands:
+  - npm install
+
+- name: unit tests
+  image: node:9
+  commands:
+  - npm run test-node
+  environment:
+    BITGOJS_TEST_PASSWORD:
+      from_secret: password
+
+- name: upload coverage
+  image: node:9
+  commands:
+  - "npx nyc report --reporter=text-lcov > coverage.lcov"
+  - "npx codecov -f coverage.lcov -B \"${DRONE_BUILD_NUMBER:-0}\" -t \"$CODECOV_TOKEN\""
+  environment:
+    CODECOV_TOKEN:
+      from_secret: codecov
+
+---
+kind: pipeline
+name: unit tests (node:10)
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: build information
+  image: node:10
+  commands:
+  - node --version
+  - npm --version
+
+- name: install
+  image: node:10
+  commands:
+  - npm install
+
+- name: unit tests
+  image: node:10
+  commands:
+  - npm run test-node
+  environment:
+    BITGOJS_TEST_PASSWORD:
+      from_secret: password
+
+- name: upload coverage
+  image: node:10
+  commands:
+  - "npx nyc report --reporter=text-lcov > coverage.lcov"
+  - "npx codecov -f coverage.lcov -B \"${DRONE_BUILD_NUMBER:-0}\" -t \"$CODECOV_TOKEN\""
+  environment:
+    CODECOV_TOKEN:
+      from_secret: codecov
+
+---
+kind: pipeline
+name: unit tests (node:11)
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: build information
+  image: node:11
+  commands:
+  - node --version
+  - npm --version
+
+- name: install
+  image: node:11
+  commands:
+  - npm install
+
+- name: unit tests
+  image: node:11
+  commands:
+  - npm run test-node
+  environment:
+    BITGOJS_TEST_PASSWORD:
+      from_secret: password
+
+- name: upload coverage
+  image: node:11
+  commands:
+  - "npx nyc report --reporter=text-lcov > coverage.lcov"
+  - "npx codecov -f coverage.lcov -B \"${DRONE_BUILD_NUMBER:-0}\" -t \"$CODECOV_TOKEN\""
```

Comment:
Shouldn't codecov just pull this from the environment? Do you have to be explicit?

Reply:
I seem to remember that this only gets set automatically for a few CI providers, and drone was not one of them. However, I'm having trouble finding that page again now. I'll take a look at it, as I'd like to add flags as well.